apps/scheduler/lib/job/pipeline/consumer.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Scheduler.Job.Consumer do
+  @moduledoc """
+  Responsible for processing jobs.
+  """
+  use GenStage
+
+  def start_link do
+    GenStage.start_link(__MODULE__, :ok)
+  end
+
+  def init(state) do
+    {:consumer, state, subscribe_to: [Scheduler.Job.ProducerConsumer]}
+  end
+
+  def handle_events(jobs, _from, state) do
+    Enum.each(jobs, fn(job) ->
+      execute_job(job)
+    end)
+
+    {:noreply, [], state}
+  end
+
+  defp execute_job(job) do
+    {m, f, a} = :erlang.binary_to_term(job.params)
+    with :ok <- Kernel.apply(m, f, a),
```

Comment:
does the scheduler have a sense of what these functions are, or do we need to add the applications that define these functions as dependencies of the scheduler?

Reply:
No we should be good here, though we should add `scheduler` as a dep in whatever app needs to implement jobs

Reply:
I tried with an example job that I put in API and it was working fine, I can check it in if helpful